src/main.cpp

```diff
@@ -9,175 +11,149 @@
-	PseudoValue* varName = new PseudoValue( rawVarName, VarName );
-	return new ASTNode{varName, &variableResolver, EMPTY_ARGS};
-}
+int main(int argc, char *argv[]) {
+	if (argc <= 1) {
+		std::cout << "No arguments has been provided" << std::endl;
+		return 1;
+	}
 
-ASTNode* createConditionalStatement(ASTNode* condition, ASTNode* trueBlock, ASTNode* falseBlock) {
-	Instructions* args = new Instructions{ condition, trueBlock, falseBlock };
-	return new ASTNode{nullptr, &conditionalStatementResolver, args};
-}
+	std::ifstream myfile;
+	myfile.open(argv[1]);
+	if (!myfile.is_open()) {
+		std::cout << "Unable to open " << argv[1] << std::endl;
+		return 1;
+	}	
 
-int main() {
 	Runtime R;
-	Instructions program = {
-		createPrint(createSum(
-			createSum(createConstInt("1"), createConstInt("15")),
-			createConstFloat("2.1")
-		)),
-		createPrint(createConstInt("69")),
-		createAssignment("a", createConstInt("6")),
-		createPrint(createGetVariable("a")),
-		createAssignment("a", createSum(createGetVariable("a"), createConstInt("1"))),
-		createPrint(createGetVariable("a")),
-		createPrint(createComparison(EQUAL, createConstString("10"), createConstInt("10"))),
-		createConditionalStatement(
-			createComparison(EQUAL, createConstString("aaa"), createConstString("aaa")),
-			createInstructionBlock(new Instructions{
-				createPrint(createConstString("TRUE")),
-			}),
-			createInstructionBlock(new Instructions{
-				createPrint(createConstString("FALSE")),
-			})
-		),
-		createForLoop(
-			createAssignment("b", createConstInt("0")),
-			createComparison(LESS, createGetVariable("b"), createConstInt("10")),
-			createAssignment("b", createSum(createGetVariable("b"), createConstInt("1"))),
-			createInstructionBlock(new Instructions{
-				createPrint(createGetVariable("b"))
-			})
-		),
-		createFunctionDeclaration(
-			"add",
-			new ASTNode{nullptr, nullptr, new Instructions{
-				new ASTNode{new PseudoValue{"a", VarName}, nullptr, EMPTY_ARGS},
-				new ASTNode{new PseudoValue{"b", VarName}, nullptr, EMPTY_ARGS},
-			}},
-			createInstructionBlock(new Instructions{
-				createPrint(createConstString("Funkcja test")),
-				createReturn(createSum(
-					createGetVariable("a"),
-					createGetVariable("b")
-				))
-			})
-		),
-		createPrint(createFunctionCall("add", new ASTNode{nullptr, nullptr, new Instructions{
-			createConstInt("2"),
-			createConstInt("3")
-		}})),
-	};
+	Instructions program =  {}; 
+
+	std::string line;
+	while (std::getline(myfile, line)) {
+		std::cout << "New line: " <<  line << std::endl;
+	 	std::vector<std::string> splitted;
+		split<std::vector<std::string>>(line, splitted);
+
+ 		int i = 0;
+		while(i < splitted.size()) {
+			// Comments implemented only for testing purposes
+			if (splitted[i] == "#")
+				break;
+
+			if (splitted[i] == "write") {
+				if (++i >= splitted.size()) {
+					throw kSyntaxError;
+				} 
+
+				ASTNode * node = getValue(i, splitted);
+				program.push_back(createPrint(node));
```

Comment:
I don't like the idea that each type has implemented addition separately. What about `write funCall() + 1` ?